CSI error releasing volume claims

### Nomad version
1.2.6

### Issue
After a sequence of instance refreshes where nomad clients were replaced by a new image the CSI volumes are in a completely unusable state. For some reason the `nomad plugin status` shows unhealthy controllers but the controller allocations report that the plugins are healthy. The Nomad server is throwing errors about it not being able to release volumes claims because:
```
    2022-03-22T12:28:58.895Z [ERROR] nomad.volumes_watcher: error releasing volume claims: namespace=mynamespace volume_id=us-west-2c-myvolume[2]
  error=
  | 1 error occurred:
  | 	* could not detach from node: Unknown node 9ef129e1-16bd-5590-85f1-2f5e5c7fd7ab
  |
```
This is happening to all volumes in the cluster. It looks like Nomad's volumes watcher does not detect disappeared (down and removed) nodes and now it's stuck trying to detach volumes from non existing nodes.

It seems the only way to workaround this is by stopping all jobs, force deregistering all volumes and reregister them.

### Reproduction steps
Setup some jobs with volumes. Refresh the clients nodes by shutting them down without draining the node first.

--- a/.changelog/12350.txt
+++ b/.changelog/12350.txt
@@ -1,0 +1,3 @@
+```release-note:bug
+csi: Fixed a bug where garbage collected nodes would block releasing a volume
+```

--- a/nomad/csi_endpoint.go
+++ b/nomad/csi_endpoint.go
@@ -1,9 +1,9 @@
 package nomad
 
 import (
-	"errors"
 	"fmt"
 	"net/http"
+	"strings"
 	"time"
 
 	metrics "github.com/armon/go-metrics"
@@ -741,7 +741,9 @@ func (v *CSIVolume) nodeUnpublishVolumeImpl(vol *structs.CSIVolume, claim *struc
 		// we should only get this error if the Nomad node disconnects and
 		// is garbage-collected, so at this point we don't have any reason
 		// to operate as though the volume is attached to it.
-		if !errors.Is(err, structs.ErrUnknownNode) {
+		// note: errors.Is cannot be used because the RPC call breaks
+		// error wrapping.
+		if !strings.Contains(err.Error(), structs.ErrUnknownNode.Error()) {
 			return fmt.Errorf("could not detach from node: %w", err)
 		}
 	}

